src/bank02.asm

```diff
@@ -8130,29 +8126,194 @@ titleScreenLicenseText:
-    TXT  "power to the one<00>"                        ;; 02:7f4f ?????????????????
-    TXT  " who touched it.<00>"                        ;; 02:7f60 ?????????????????
-    TXT  "<00>"                                        ;; 02:7f71 ?
-    TXT  "  Dark Lord was<00>"                         ;; 02:7f72 ????????????????
-    TXT  "  trying to find<00>"                        ;; 02:7f82 ????????????????.
-    TXT  "  the way to the<00>"                        ;; 02:7f93 ...........??????
-    TXT  " Tree of Mana to<00>"                        ;; 02:7fa4 ?????????????????
-    TXT  "  get the mighty<00>"                        ;; 02:7fb5 ?????????????????
-    TXT  " power to conquer<00>"                       ;; 02:7fc6 ??????????????????
-    TXT  "    the world.<00>"                          ;; 02:7fd8 ???????????????
-    TXT  "<00>"                                        ;; 02:7fe7 ?
-    db   $00, $00, $00, $00, $00, $00, $00, $00        ;; 02:7fe8 ????????
-    db   $00, $00, $00, $00, $01, $ff, $ff, $ff        ;; 02:7ff0 ????????
-    db   $ff, $ff, $ff, $ff, $ff, $ff, $ff, $ff        ;; 02:7ff8 ????????
+    TXT  "<00>"                                        ;; 02:7fd8 ?
+    db   $01                                           ;; 02:7fd9 ?
+
+; Draws left aligned number HL as modified tiles at WRAM position DE
+; C holds the drawing mode in the first nibble (0 for shift, 1 for shift/swap)
+; C holds number of max digits in the second nibble
+drawLeftAlignedNumberInWRAM:
+    ld   B, $00 ; digit counter
+
+    ; Housekeeping, get a copy of proper WRAM starting
+    ;  location on the stack for later use
+    push BC
+    push DE
+
+    ; Load up the digits on the stack
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
+    pop  DE
+    pop  BC
+    push AF
+    inc  B
+    ld   A, H
+    or   A, L
+    jr   NZ, .divmod_and_store_number
+
+    ; Reduce second nibble of C by num digits found
+    ; Remainder provides the number of tiles to clear
+    ; after we write the digits
+    ld   A, C
+    sub  A, B
+    ld   C, A
+
+    ; If in swap mode, clear the first tile and increment the
+    ; tile buffer pointer in DE. The upcoming swap routine
+    ; will fill the first tile in with 4px from the next tile
+    and  A, $10
+    jr   Z, .swap_test_skip
+
+    ld   H, D
+    ld   L, E
+    xor  A, A
+    call memsetTileWithA ; increments HL by $10
+    ld   D, H
+    ld   E, L
+ .swap_test_skip:
+
+    ; Stack holds B digits to pop, transfer them
+ .grab_digit_and_transfer:
+    ld   HL, gfxStatusBar+$100 ; start of number tiles
+    pop  AF ; grab digit
+    swap A
+    push DE
+    ld   D, $00
+    ld   E, A
+    add  HL, DE
+    pop  DE
+    ld   A, BANK(gfxStatusBar)
+    push BC
+    ld   B, $10
+    call copyAndRotateBankAfromHLtoDE
+    pop  BC
+    dec  B
+    jr   NZ, .grab_digit_and_transfer
+
+    ; Clear remaining tiles
+    ld   A, C
+    and  A, $0f
+    jr   Z, .check_swap
+
```

Comment:
What about if in swap mode: 
1) Push the real first tile address above (in the code just before `. swap_test_skip`) and `inc C`
2) If swap mode pop the address here.

I think that would (work and also) allow eliminating the tile clear before `.swap_test_skip`?

Reply:
I had used the stack to push the digits (from right to left), so I can't push anything else on the stack until I pop them back off (from left to right).

Still, I like the idea of using the stack to push the addresses to clear. I think there's a solution in there to have the tile clearing all in one spot.

It's hard for me to get much coding time during the week, but I may be able to get to it this evening.

Reply:
Good point. I'm probably overlooking things like that because I'm not testing it yet. I think I'd like to land this soon so I can easily test and edit it, but I'm busy today and tonight.

On an implementation note, using the stack seems weird, but I assume that's to avoid having to allocate memory. Several functions--including the BCD conversion--use the beginning of the area where save games are initially loaded into SRAM as scratch. Look at `wSRAMSaveHeader`--there are some notes there.

Reply:
Ok, restructured a few things to make it cleaner/faster.
1) Swap routine now goes from right to left instead of left to right
2) Tile clearing occurs in one spot, so I inlined that memset code
3) Saved a couple instructions in the grab_digit_and_transfer loop